### PR TITLE
Reduce overhead by fixing alignment in SubsetLb

### PR DESCRIPTION
**Pull request by DiazAlan**
Commit Message: Reduce overhead by fixing alignment in SubsetLb
Additional Description: Moves many of the smaller members of the class to the end to save small amounts of memory
Risk Level: Low

### CODE REVIEW

File: source/common/upstream/subset_lb.h
Comment (adisuissa) [inline source/common/upstream/subset_lb.h:430]: nit: please add a small note about the coalescing of these data members at the end (something like [this](https://github.com/envoyproxy/envoy/blob/1ac0a5948ca317fd418777dab2ffe27935c0a276/source/common/router/config_impl.h#L342)).

**Review by adisuissa — COMMENTED** [review]
LGTM, modulo small nit.
Thanks!

File: source/common/upstream/subset_lb.h
Comment (DiazAlan) [reply source/common/upstream/subset_lb.h:431]: Done

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

/assign-from @envoyproxy/senior-maintainers

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**
